evm/vm/base.py

```diff
@@ -209,7 +209,10 @@ def state_in_temp_block(self):
         header = self.block.header
         temp_block = self.generate_block_from_parent_header_and_coinbase(header, header.coinbase)
         prev_hashes = (header.hash, ) + self.previous_hashes
-        state = self.get_state(block_header=temp_block.header, prev_hashes=prev_hashes)
+        state = self.get_state(self.chaindb, temp_block, prev_hashes)
+        if state.gas_used > 0:
+            raise Exception("There must not be any gas used in a fresh temporary block")
```

Comment:
Maybe `ValidationError`?

Reply:
I've been thinking about adding an `Invariant` exception for these types of things where what we are really doing is assert.